validation-test/stdlib/UnsafeBufferPointerSlices.swift

```diff
@@ -1,5 +1,5 @@
 // RUN: %target-run-simple-swift
-// REQUIRES: executable_test
+// REQUIRES: ( executable_test && !target={{(amd64|x86_64|x86_64h)-.*}} ) || ( executable_test && !asan )
 // END.
```

Comment:
Please express this as:
`UNSUPPORTED: target=...`
* I think this is easier to understand
* Easier to change (pure addition instead of line modification)

Please also state the radar number here that's tracking the fix.

Reply:
I added rdar://106875095 to the commit

About UNSUPPORTED - I wasn't sure about what he correct solution is, the LLVM testing docs ( https://llvm.org/docs/TestingGuide.html ) don't  elaborate about logic executed when the multiple REQURES/UNSUPPORTED statements are present - in particular they say nothing about precedence of those. What happens when contradicting REQUIRES/UNSUPPORTED statements are present?  

Is the decision on whether to execute the test the AND of the both (multiple) conditions ( R && !U )? Couldn't figure that out, so I went with a single logical expression, generating the truth table I want - i.e. the one that  preserves the current logic ( executable_test ), and adds two extra inputs - CPU kind and asan.
 
The entire expression does what I want - runs the tests normally on all CIs, except the asan CI when CPU is x86_64.* - and is compliant with the LLVM testing docs.  

Besides, there are examples in LLVM codebase using same approach (single REQUIRES with multiple inputs), for instance:  https://github.com/llvm/llvm-project/blob/main/lldb/test/Shell/Watchpoint/netbsd-nouserdbregs.test#L4 

So I'd rather keep it as it is.

The output of the failing test - well, the test times out, hence I couldn't really get any output out of it

Reply:
Hmm, again, I would strongly before something that follows this scheme:
```
// REQUIRES: executable_test

// rdar://106875095
// UNSUPPORTED: asan && target={{(amd64|x86_64|x86_64h)-.*}}

// next radar
// UNSUPPORTED: <next disablement condition>
```

That's easier to manage over time: add / remove disablements.

Combinations of requires & unsupported should be possible and LIT should do the right thing?!  I would be surprised to learn if it didn't.

For example, I see tests with:
```
// REQUIRES: executable_test
// REQUIRES: concurrency
// UNSUPPORTED: use_os_stdlib
// UNSUPPORTED: back_deployment_runtime
```

Reply:
Can the unsupported directive be simplified to this?
```
// rdar://106875095 - Swift CI: test times out with asan
// UNSUPPORTED: asan
```